### ISSUE TITLE
Provide a way to avoid using cache

### ISSUE THREAD

**Issue by Austreelis** [packwiz/packwiz#129]
I'm trying to package a modpack using [nix](nixos.org), which requires the "build" (basically a `packwiz mr export --pack-file $packFile -o $out`) to be completely sandboxed (barring network access).

To enforce this, nix replaces the build user home directory with the non-existing `/homeless-shelter` which avoid the user cache directory to exist:

```
Loading modpack...
Refreshing index... 100 %done
Reading external files...
Retrieving 83 external files...
Disclaimer: you are responsible for ensuring you comply with ALL the licenses, or obtain appropriate permissions, for the files "added to zip" below
packwiz is currently unable to match metadata between mod sites - if any of these are available from Modrinth you should change them to use Modrinth metadata (e.g. by reinstalling them using the mr commands)

Error retrieving external files: failed to create cache directory: mkdir /homeless-shelter: permission denied
```
([Nix build file generating this error](https://gitlab.com/austreelis/tuxcraft/-/blob/main/nix/tuxcraft-mrpack.nix), alternatively `nix build gitlab:austreelis/tuxcraft`)

Caching being not desirable in a nix build (as nix tries really hard to enforce reproducible builds, hence the sandboxing), I'd be happy with a `--no-cache` option that avoids using the cache.

I'd happily work on a PR :)

**Comment by comp500** [comment]
Hmm... surely if network access is barred, the export wouldn't be able to occur anyway? `packwiz` needs to download files to retrieve their hashes, since Modrinth requires `sha1`/`sha512` hashes and the pack format currently can't store more than one hash.

I'll accept a `--no-cache` PR but note that the caching code depends on writing to a file; it may be better to just change the cache directory to a temporary one?

**Comment by Austreelis** [comment]
> surely if network access is barred, the export wouldn't be able to occur anyway?

Sorry if that was unclear, I meant network access is more or less the only hole in the sandbox.

To be clear, the only thing I have access to is a writable temp directory with the project's source, a writable output directory, network access and private versions of `/proc`, `/dev`, `/dev/shm` and `/dev/pts`.

> note that the caching code depends on writing to a file; it may be better to just change the cache directory to a temporary one ?

If you think that's a better way to go, that works for me. Am I right that it would mean controlling the cache directory with something like a `--cache-dir <DIR>` ?

**Comment by comp500** [comment]
Yeah, that'd work quite well! There's functions for getting the cache directory that you can just check the viper config from.

**Comment by comp500** [comment]
Fixed by #133.
